### PR TITLE
View Settings Facet (1578)

### PR DESCRIPTION
**Pull request by Flaurite**
### Breaking changes

1. `UiSetting` entity: renamed property `name` -> `key`
2. Renamed interface `UserSettingService` (classic) -> `UserSettingsService`
3. Changed methods in `UserSettingsService`:

|Classic | FlowUI|
|--------|--------|
|`String loadSetting(String name)`                                              | `Optional<String> load(String key)`                            |
|`void saveSetting(String name, @Nullable String value)`          |`void save(String key, @Nullable String value)`            |
|`void deleteSettings(String name)`                                            |`void delete(String key)`                                                |
|`void copySettings(UserDetails fromUser, UserDetails toUser)`|`void copy(String fromUsername, String toUsername)`|
|`void deleteScreenSettings(Set<String> screens);`                   | - |

4. `UiSettingsCache` -> `UserSettingsCache` and changed methods, but it's internal API, so it should be ok.

### CODE REVIEW

File: jmix-flowui/flowui-data/src/main/java/io/jmix/flowuidata/entity/UiSetting.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+import org.springframework.data.annotation.CreatedDate;
+
+import java.io.Serializable;
+import java.util.Date;
+import java.util.UUID;
+
+/**
+ * Record for user's UI setting.
+ */
+@Entity(name = "flowui_UiSetting")
```
Comment (glebfox) [inline jmix-flowui/flowui-data/src/main/java/io/jmix/flowuidata/entity/UiSetting.java:32]: 1. `ui_`
2. `UI_`

File: jmix-flowui/flowui-data/src/main/java/io/jmix/flowuidata/settings/UserSettingsCacheImpl.java [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.lang.Nullable;
+import org.springframework.stereotype.Component;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+@Component("flowui_UserSettingsCacheImpl")
```
Comment (glebfox) [inline jmix-flowui/flowui-data/src/main/java/io/jmix/flowuidata/settings/UserSettingsCacheImpl.java:33]: ui_

**Review by glebfox — APPROVED** [review]

File: jmix-flowui/flowui-data/src/main/resources/io/jmix/flowuidata/liquibase/changelog/001-flowui-data.xml [hunk 3]
```diff
@@ -54,4 +54,17 @@
                       columnDataType="varchar(255)"/>
     </changeSet>
 
+    <changeSet author="flowui-data" id="4">
+        <createTable tableName="FLOWUI_UI_SETTING">
```
[the remark this answers is not in the document]
Comment (Flaurite) [reply jmix-flowui/flowui-data/src/main/resources/io/jmix/flowuidata/liquibase/changelog/001-flowui-data.xml]: Fixed

File: jmix-flowui/flowui-data/src/main/java/io/jmix/flowuidata/settings/UserSettingsCacheImpl.java (hunk 2 above)
Comment (Flaurite) [reply jmix-flowui/flowui-data/src/main/java/io/jmix/flowuidata/settings/UserSettingsCacheImpl.java:33]: Fixed

File: jmix-flowui/flowui-data/src/main/java/io/jmix/flowuidata/settings/UserSettingsRemoveUserListener.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+import jakarta.persistence.PersistenceContext;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.transaction.event.TransactionPhase;
+import org.springframework.transaction.event.TransactionalEventListener;
+
+import java.util.List;
+
+@Internal
+@Component("flowui_UserSettingsRemoveUserListener")
```
[the remark this answers is not in the document]
Comment (Flaurite) [reply jmix-flowui/flowui-data/src/main/java/io/jmix/flowuidata/settings/UserSettingsRemoveUserListener.java]: Fixed

File: jmix-flowui/flowui-data/src/main/java/io/jmix/flowuidata/settings/ViewSettingsComponentManagerImpl.java [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+import io.jmix.flowui.facet.settings.ViewSettingsComponentRegistry;
+import io.jmix.flowui.facet.settings.component.binder.ComponentSettingsBinder;
+import io.jmix.flowui.facet.settings.component.binder.DataLoadingSettingsBinder;
+import io.jmix.flowui.settings.UserSettingsCache;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collection;
+
+@org.springframework.stereotype.Component("flowui_ViewSettingsComponentManagerImpl")
```
[the remark this answers is not in the document]
Comment (Flaurite) [reply jmix-flowui/flowui-data/src/main/java/io/jmix/flowuidata/settings/ViewSettingsComponentManagerImpl.java]: Fixed in all new beans and entities.

File: jmix-flowui/flowui/src/main/java/io/jmix/flowui/facet/settings/component/SimplePaginationSettings.java [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.jmix.flowui.facet.settings.component;
+
+import io.jmix.flowui.facet.settings.Settings;
+import org.springframework.lang.Nullable;
+
+public class SimplePaginationSettings implements Settings {
```
[the remark this answers is not in the document]
Comment (Flaurite) [reply jmix-flowui/flowui/src/main/java/io/jmix/flowui/facet/settings/component/SimplePaginationSettings.java:22]: Reworked as discussed previously: URL query parameters should have priority over UiSettings.

**Pull request merged**